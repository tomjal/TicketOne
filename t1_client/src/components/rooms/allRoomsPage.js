import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { getAllRooms } from './../../actions/roomsActionCreators';
import { getAllMessages, sendMessageToRoom } from './../../actions/messagesActionCreators';

import RoomWidget from './roomWidget';

const MAX_FLEX_ELEMS_IN_ROW = 3;

const mapStateToProps = (state) => {
  return {
    id: state.context.id,
    role: state.context.role,
    rooms: state.rooms,
    messages: state.messages
  }
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getAllRooms,
    getAllMessages,
    sendMessageToRoom
  }, dispatch);
};

export class AllRoomsPage extends Component {
  constructor(props) {
    super(props)
    this.sendMessageCallback = this.sendMessageCallback.bind(this);
  }
  componentDidMount() {
    this.props.getAllRooms();
    this.props.getAllMessages();
  }
  sendMessageCallback(message, id) {
    this.props.sendMessageToRoom(message, id, this.props.role, this.props.id);
  }
  getEmptyFlexFillers(numbInRow, numbOfAll) {
    let emptyElemsNumber = numbInRow - (numbOfAll % numbInRow);
    let emptyFlexElems = [];
    for (let i = 0; i < emptyElemsNumber; i++) {
      emptyFlexElems.push({ id: i });
    }
    return emptyFlexElems;
  }
  render() {
    const { role, rooms, messages } = this.props;
    const emptyFlexElems = this.getEmptyFlexFillers(MAX_FLEX_ELEMS_IN_ROW, rooms.length);

    return (
      <div>
        {rooms.length !== 0 &&
          <div className="room-flex-container">
            {rooms.map((roomId, i) => {
              let localMessages = [];
              if (messages[roomId]) {
                localMessages = messages[roomId];
              }
              return <div key={i}>
                <RoomWidget
                  id={roomId}
                  role={role}
                  sendMessageCallback={this.sendMessageCallback}
                  messages={localMessages} />
              </div>;
            })}
            {emptyFlexElems.map((roomId, i) => {
              return <div key={`filler-${i}`} className="room-flex-container__filling"></div>;
            })}
          </div>
        }
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AllRoomsPage);
