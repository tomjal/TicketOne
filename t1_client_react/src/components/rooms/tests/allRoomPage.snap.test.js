import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { AllRoomsPage } from './../allRoomsPage';

test('AllRoomsPage snapshot', () => {
  const props = {
    rooms: [],
    getAllRooms: jest.fn(),
    getAllMessages: jest.fn()
  };
  const wrap = shallow(<AllRoomsPage {...props} />);

  expect(shallowToJson(wrap)).toMatchSnapshot();
});