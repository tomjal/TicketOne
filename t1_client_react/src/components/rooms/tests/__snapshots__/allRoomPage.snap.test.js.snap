// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllRoomsPage snapshot 1`] = `<div />`;
